warehouse/router/router_test.go

```diff
@@ -1112,6 +1108,7 @@ func TestRouter(t *testing.T) {
 		}))
 
 		ctx, cancel := context.WithCancel(context.Background())
+		g, _ := errgroup.WithContext(ctx)
```

Comment:
Why not using `errgroup.Group` if you're ignoring the context? Internally the context is not stored nor used at all, it is just for the end user:

```go
func WithContext(ctx context.Context) (*Group, context.Context) {
	ctx, cancel := withCancelCause(ctx)
	return &Group{cancel: cancel}, ctx
}
```

If one of the routines errors then the context is going to be cancelled but since you're ignoring it then:
```suggestion
		g := errgroup.Group{}
```